### PR TITLE
Upcoming events section

### PR DESCRIPTION
**Pull request by shami-dev**
## PR Title
Implement Responsive Cards Section

## Summary
Create a responsive cards section for the home page using Figma mockup. 

Play and do reusable buttons using Class Variance Authority (CVA). 

Add a button component from the Reakit library. It follows the WAI-ARIA Button Pattern. For example, the button should respond to Enter and Space keys.

Add reusable cards. 

## Details (reason and description of the changes)
This PR introduces a navigation feature to the upcoming events section, enabling users to easily access and explore the upcoming events. 

Try to make reusable buttons as if we have a design system. In the design system we tend to keep consistent all the elements such as colors, text sizes, paddings, etc. I picked CVA because it gives an easy-to-use interface to define variants that conditionally apply class sets. Simply put, if we have a bunch of props that, for example, "button" or more complex elements consume, we can make them easy to read and type safely using CVA. Also, it helps to avoid conflicts. 

It was easy to import this component and use it with CVA. Reakit's button looks like the previous buttons. That's why images are not added. 

Transform cards into cards that can consume props. 

## How has the UI changed?
This is the homepage screenshot on a laptop:
Laptop-Cards

Buttons appearance on a laptop:
Laptop-Buttons

This is the homepage screenshot on iPhone SE1:
iPhone-SE-Cards

Buttons appearance on a iPhone SE1:
iPhone-SE-Buttons

The second card has changed because of mapping the card. 
Reusable-cards

## What tests have been added?
No tests have been completed

## If a new npm package was added, why was it chosen and why was it necessary?
No new npm package was installed

class-variance-authority

reakit

### CODE REVIEW

File: app/components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { cva, type VariantProps } from "class-variance-authority";
```
Comment (andrelandgraf) [inline app/components/Button.tsx:1]: Could you add a small description and motivation to the PR description as to why you picked this library and what we will use it for?

File: package.json [hunk 2]
```diff
@@ -7,21 +7,28 @@
     "typecheck": "tsc"
   },
   "dependencies": {
+    "@ariakit/react": "^0.2.3",
+    "@radix-ui/react-label": "^2.0.1",
     "@remix-run/node": "^1.7.2",
     "@remix-run/react": "^1.7.2",
     "@remix-run/vercel": "^1.7.2",
+    "@remix-run/web-fetch": "^4.3.4",
```
Comment (andrelandgraf) [inline package.json:15]: When creating a PR, please be mindful of what's in your `package.json`. I don't think this package is used anywhere?

File: package.json [hunk 3]
```diff
@@ -7,21 +7,28 @@
     "typecheck": "tsc"
   },
   "dependencies": {
+    "@ariakit/react": "^0.2.3",
+    "@radix-ui/react-label": "^2.0.1",
```
Comment (andrelandgraf) [inline package.json:11]: Are these packages in use?

File: package.json [hunk 4]
```diff
@@ -7,21 +7,28 @@
...
     "@remix-run/react": "^1.7.2",
     "@remix-run/vercel": "^1.7.2",
+    "@remix-run/web-fetch": "^4.3.4",
     "@vercel/node": "^2.10.3",
+    "class-variance-authority": "^0.6.0",
     "isbot": "^3.6.8",
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "tailwind-merge": "^1.12.0"
```
Comment (andrelandgraf) [inline package.json:21]: Same here!

**Comment by CamilingJS** [comment]
Aesthetically I don't believe this library design matches the theme of the current design. Would you agree?

File: app/components/Button.tsx (hunk 1 above)
Comment (ShowMale) [reply app/components/Button.tsx:1]: Sure. Added.

File: package.json (hunk 2 above)
Comment (ShowMale) [reply package.json:15]: This package helped me with an error related to my Node version (v.20) which appeared after the "npm run dev" command. It seems that all project members should use the same Node versions. I can add it to .gitignore.

File: package.json (hunk 3 above)
Comment (ShowMale) [reply package.json:11]: Not yet) As we still didn't define the library for components, I left them here to import and try the other elements from Ariakit and Radix. Anyway, I can delete them all.

**Pull request closed without merging**